### PR TITLE
Add support for filter operators

### LINKED ISSUE #105: Add support for filter operators
# Purpose of the Issue
I would like to add support for filtering operators in Soul. Currently, Soul filters table fields with the `_filters` field, which works fine, but I am unable to filter items using the operators `<`, `>`, `<=`, `>=`, and `!=`. For example, I can fetch an invoice from Soul using the following URL, which works fine:

   ```js
      localhost:8000/api/tables/invoices/rows?_filters=Total:1.98
   ```

However, I might want to fetch invoices where the total value is less than 2 or greater than 2, and I cannot achieve this with Soul's current functionality.

# Proposed Solution
There could be several options to add this feature, but I propose passing double underscores after the column name. Here are some examples:

**Greater than (`>`):** 
```js
localhost:8000/api/tables/invoices/rows?_filters=Total__gt:3
```

**Less than (`<`):**
```js
localhost:8000/api/tables/invoices/rows?_filters=Total__lt:3
```

**Greater than or equal to (`>=`):**
```js
localhost:8000/api/tables/invoices/rows?_filters=Total__gte:3
```

**Less than or equal to (`<=`):**
```js
localhost:8000/api/tables/invoices/rows?_filters=Total__lte:3
```

**Not equal to (`!=`):**
```js
localhost:8000/api/tables/invoices/rows?_filters=Total__neq:3
```

This proposal would allow users to filter table rows using the operators listed above, making Soul's functionality more flexible and powerful.

### PR DESCRIPTION
**Pull request by AbegaM**
Fixes #105 

# Purpose of PR

The purpose of this PR is to enhance the filtering feature of `Soul`. Currently, the `_filters` field is used for basic filtering, but it does not allow users to filter items by comparing them or by using comparison operators.

# Modifications

To address this limitation, I have made modifications to the `rows.js` file in the `core/src/controllers` directory. With these changes, users can now filter items using comparison operators such as `__lte`, `__gte`, `__lt`, `__gt`, `__eq`, and `__neq`. With this improved filtering functionality, users can more effectively search and analyze data in the Soul application.

### CODE REVIEW

**Comment by AbegaM** [comment]
Hello @thevahidal Please review this PR and let me know if there is anything i have to improve

File: core/src/controllers/rows.js [hunk 1]
```diff
@@ -66,15 +75,23 @@ const listTableRows = async (req, res) => {
   // will filter by name like '%John%' and age like '%20%'
 
   const filters = _filters.split(',').map((filter) => {
-    const [field, value] = filter.split(':');
-    return { field, value };
+    let [key, value] = filter.split(':');
```
Comment (IanMayo) [inline core/src/controllers/rows.js:78]: I'd like this to be more pessimistic, please @AbegaM .  I'd like us to handle when a filter doesn't match our expected list, and throw a suitable error message.

Comment (AbegaM) [reply core/src/controllers/rows.js:78]: @IanMayo Ah, I haven't considered that. I will modify the code to throw errors.

Comment (AbegaM) [reply core/src/controllers/rows.js]: @thevahidal, I have added error handling to the feature request. Please review it and let me know if there is anything I can modify.

**Review by IanMayo — APPROVED** [review]
Have tested this functionality using calls from browser, and inspecting results.

Found issues with `next/prev` links, but these have been captured in #109

**Comment by thevahidal** [comment]
Thanks for the effort guys, it looks good to go

**Pull request merged**